apis/python/src/tiledbsoma/indexed_dataframe.py

```diff
@@ -200,18 +201,46 @@ def read(
                 query_condition=query_condition,
             )
 
+            # Acceptable ways to index:
+            # * None
+            # * A sequence of coordinates is accepted, one per dimension.
+            # * Sequence length must be at least one and <= number of dimensions.
+            # * If the sequence contains missing coordinates (length less than number of dimensions),
+            #   then "slice(None)" is assumed for the missing dimensions.
+            # * Per-dimension, explicitly specified coordinates can be one of: None, a value, a
+            #   list/ndarray/paarray/etc of values, a slice, etc.
+
             if ids is not None:
-                dim_name = A.schema.domain.dim(0).name
-                if isinstance(ids, list):
-                    sr.set_dim_points(dim_name, ids)
-                elif isinstance(ids, pa.ChunkedArray):
-                    sr.set_dim_points(dim_name, ids)
-                elif isinstance(ids, pa.Array):
-                    sr.set_dim_points(dim_name, pa.chunked_array(ids))
-                elif isinstance(ids, slice):
-                    lo_hi = util.slice_to_range(ids)
-                    if lo_hi is not None:
-                        sr.set_dim_ranges(dim_name, lo_hi)
+                if not (isinstance(ids, list) or isinstance(ids, tuple)):
+                    raise SOMAError(
+                        f"ids type {type(ids)} unhandled; expected list or tuple"
+                    )
+                if len(ids) < 1 or len(ids) > A.schema.domain.ndim:
+                    raise SOMAError(
+                        f"ids {ids} must have length between 1 and ndim ({A.schema.domain.ndim}); got {len(ids)}"
+                    )
+
+                for i in range(len(ids)):
+                    dim_name = A.schema.domain.dim(i).name
+                    dim_ids = ids[i]
+                    if dim_ids is None:  # select all in this dimension
+                        pass
+                    elif isinstance(dim_ids, int):
+                        sr.set_dim_points(dim_name, [dim_ids])
+                    elif isinstance(dim_ids, list):
```

Comment:
should also support tuple, i,e, `instance(dim_ids, (list, tuple))`

Reply:
Also you may find it beneficial to use the multi-type version of `instance`, which would allow many of these to be done on a single line, eg,

```
elif isinstance(dim_ids, (tuple, list, pa.ChunkedArray)):
    sr.set_dim_points(dim_name, dim_ids)
```